site/src/lib/api/blocks/r2.ts

```diff
@@ -18,13 +24,105 @@ const s3 = new S3({
   signatureVersion: "v4",
 });
 
+const stripLeadingAt = (pathWithNamespace: string) =>
+  pathWithNamespace.replace(/^@/, "");
+
+const validateBlockFiles = async (
```

Comment:
here is where the validation code that was previously in `npm` has been moved to